File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Deque<String> dq = new ArrayDeque<>();

        int index = 0;
        for(char ch: s.toCharArray()){
            if(index ==0 && ch == ')'){
                return false;
            }
            
            if(ch == '('){
                dq.push(String.valueOf(ch));
            } else{
                if(dq.isEmpty()){
                    return false;                    
                }
                dq.pop();
            }
            index++;
        }
        
        if(dq.size()>0){
            return false;
        } else{
            return true;
        }
        
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    let stack = [];
    
    let arr = s.split('');
    

    arr.forEach(elem => {
        if(elem == '('){
            stack.push(elem)
        } else if(elem == ')' && stack[stack.length-1]== '(' ){
            stack.pop();
        } else if( elem == ')'){
            stack.push(elem);
        }
    })
    
    if(stack.length==0){
        return true
    } else{
        return false
    }
        
    return answer;
}